model/main.py

```diff
@@ -27,31 +28,35 @@ def parse_args() -> argparse.Namespace:
     return argparser.parse_args()
 
 
-def main(args):
-    mlflowhandler = MLFlowHandler()
-    df = preprocess_data(mlflowhandler, pd.read_csv(args.train_data))
+def main(args: argparse.Namespace) -> None:
+    uri = os.environ.get("MLFLOW_TRACKING_URI", "http://48.209.80.111:5000")
+    experiment_name = os.environ.get("MLFLOW_EXPERIMENT_NAME", "housepricesml")
+    mlflow_handler = MLFlowHandler(uri, experiment_name)
+    df = pd.read_csv(args.train_data)
+    uncorrelated_col, insignificant_col, missing_values_col = get_data_for_preprocessing(
+        df, treshhold=0.05, mlflow=mlflow_handler
+    )
+    df = preprocess_data(df, uncorrelated_col, insignificant_col, missing_values_col)
+    mlflow_handler.log_preprocessed_data(df)
     # df = normalize_data(df)
     X_train, X_test, y_train, y_test = split_data(df)
     model_type = ModelType(args.model)
     if model_type.value == ModelType.REGRESSOR.value:
         model = Regressor()
-        params = model.model.get_params()
     elif model_type.value == ModelType.XGB.value:
         with open(args.config, "r") as f:
             params = json.load(f)
         model = XGBModel(**params)
-        params = model.model.get_params()
     else:
         model = NeuralNetwork()
         model.add_to_model(Dense(64, activation="relu"))
         model.add_to_model(Dense(32, activation="relu"))
         model.add_to_model(Dense(1, activation="linear"))
         model.compile()
-        params = model.model.get_config()
     model.fit(X_train, y_train)
     score = model.evaluate(X_test, y_test)
-    mlflowhandler.log_model(model, ("r2_score", score), params, args.train_data, X_train)
-    mlflowhandler.close()
+    mlflow_handler.log_model(model, ("r2_score", score), X_train, model.get_params(), args.train_data)
+    mlflow_handler.close()
```

Comment:
This is just a suggestion. You can implement a context manager and use `with` statement. Then you have certainty that resources are freed after usage and don't have to remember about having `.close()` method call after.